api/gateway/apimiddleware/api_middleware.go

```diff
@@ -74,97 +76,117 @@ type fieldProcessor struct {
 // Run starts the proxy, registering all proxy endpoints.
 func (m *ApiProxyMiddleware) Run(gatewayRouter *mux.Router) {
 	for _, path := range m.EndpointCreator.Paths() {
-		m.handleApiPath(gatewayRouter, path, m.EndpointCreator)
+		m.handleApiPath(gatewayRouter, path)
 	}
 }
 
-func (m *ApiProxyMiddleware) handleApiPath(gatewayRouter *mux.Router, path string, endpointFactory EndpointFactory) {
-	gatewayRouter.HandleFunc(path, func(w http.ResponseWriter, req *http.Request) {
-		endpoint, err := endpointFactory.Create(path)
-		if err != nil {
-			errJson := InternalServerErrorWithMessage(err, "could not create endpoint")
-			WriteError(w, errJson, nil)
+func (m *ApiProxyMiddleware) handleApiPath(gatewayRouter *mux.Router, path string) {
+	gatewayRouter.HandleFunc(path, m.HandleFunc)
+}
+
+func (m *ApiProxyMiddleware) HandleFunc(w http.ResponseWriter, req *http.Request) {
+	endpoint, err := m.EndpointCreator.Create(req.URL.Path)
+	if err != nil {
+		errJson := InternalServerErrorWithMessage(err, "could not create endpoint")
+		WriteError(w, errJson, nil)
+	}
+
+	for _, handler := range endpoint.CustomHandlers {
+		if handler(m, *endpoint, w, req) {
+			return
 		}
+	}
 
-		for _, handler := range endpoint.CustomHandlers {
-			if handler(m, *endpoint, w, req) {
-				return
-			}
+	if req.Method == "POST" {
+		if errJson := deserializeRequestBodyIntoContainerWrapped(endpoint, req, w); errJson != nil {
+			WriteError(w, errJson, nil)
+			return
 		}
 
-		if req.Method == "POST" {
-			if errJson := deserializeRequestBodyIntoContainerWrapped(endpoint, req, w); errJson != nil {
-				WriteError(w, errJson, nil)
-				return
-			}
+		if errJson := ProcessRequestContainerFields(endpoint.PostRequest); errJson != nil {
+			WriteError(w, errJson, nil)
+			return
+		}
+		if errJson := SetRequestBodyToRequestContainer(endpoint.PostRequest, req); errJson != nil {
+			WriteError(w, errJson, nil)
+			return
+		}
```

Comment:
`HandleFunc` is getting a little long, I'd consider refactoring code blocks under conditions `req.Method==""POST`, `req.Method == "DELETE"` and `! GrpcResponseIsEmpty()` into own methods